app/screens/StateScreen.tsx

```diff
@@ -39,6 +50,72 @@ export function StateScreen() {
     }))
   }
 
+  const createSnapshot = () => {
+    if (!activeTab) {
+      console.log("No active client to create snapshot from")
+      return
+    }
+    sendToCore("state.backup.request", { clientId: activeTab })
+  }
+
+  const copySnapshotToClipboard = (snapshot: Snapshot) => {
+    try {
+      IRClipboard.setString(JSON.stringify(snapshot.state, null, 2))
+      console.log("Snapshot copied to clipboard")
+    } catch (error) {
+      console.error("Failed to copy snapshot to clipboard:", error)
+    }
+  }
+
+  const copyAllSnapshotsToClipboard = () => {
+    try {
+      console.log("Copying all snapshots to clipboard", snapshots)
+      IRClipboard.setString(JSON.stringify(snapshots, null, 2))
+      console.log("All snapshots copied to clipboard")
+    } catch (error) {
+      console.error("Failed to copy snapshots to clipboard:", error)
+    }
+  }
```

Comment:
This can be extracted to a utility function as well. You can either pass in the `snapshots` to the function or just get them on-demand with `withGlobal` (or if @SeanBarker182 has renamed that function, whatever he landed on).